packages/api/src/beacon/server/events.ts

```diff
@@ -18,7 +18,13 @@ export function getRoutes(config: ChainForkConfig, api: ServerApi<Api>): ServerR
         const controller = new AbortController();
 
         try {
-          // Add injected headers from other pluggins. This is required for fastify-cors for example
+          // Prevent Fastify from sending the response, this is recommended before writing to the `.raw` stream
+          // and avoids "Cannot set headers after they are sent to the client" errors during shutdown or abrupt client aborts.
+          // See https://github.com/fastify/fastify/issues/3979, https://github.com/ChainSafe/lodestar/issues/5783
+          // eslint-disable-next-line @typescript-eslint/no-floating-promises
```

Comment:
`hijack()` is not async, see [reply.js#L116](https://github.com/fastify/fastify/blob/b31b1b48a148e8fae4d65a7b8eb2d271389d02db/lib/reply.js#L116)